samples/sensor/dht_polling/src/main.c

```diff
@@ -32,29 +50,49 @@ int main(void)
 
 	while (1) {
 		for (size_t i = 0; i < ARRAY_SIZE(sensors); i++) {
-			struct device *dev = (struct device *)sensors[i];
+			struct device *dev = (struct device *) sensors[i];
+
+			uint8_t buf[128];
 
-			rc = sensor_sample_fetch(dev);
-			if (rc < 0) {
-				printk("%s: sensor_sample_fetch() failed: %d\n", dev->name, rc);
+			rc = sensor_read(dht_iodev[i], &dht_ctx, buf, 128);
+
+			if (rc != 0) {
+				printk("%s: sensor_read() failed: %d\n", dev->name, rc);
 				return rc;
 			}
 
-			struct sensor_value temp;
-			struct sensor_value hum;
+			const struct sensor_decoder_api *decoder;
+
+			rc = sensor_get_decoder(dev, &decoder);
 
-			rc = sensor_channel_get(dev, SENSOR_CHAN_AMBIENT_TEMP, &temp);
-			if (rc == 0) {
-				rc = sensor_channel_get(dev, SENSOR_CHAN_HUMIDITY, &hum);
-			}
 			if (rc != 0) {
-				printf("get failed: %d\n", rc);
-				break;
+				printk("%s: sensor_get_decode() failed: %d\n", dev->name, rc);
+				return rc;
 			}
 
-			printk("%16s: temp is %d.%02d °C humidity is %d.%02d %%RH\n",
-							dev->name, temp.val1, temp.val2 / 10000,
-							hum.val1, hum.val2 / 10000);
+			uint32_t temp_fit = 0;
+			uint8_t temp_buf[64];
```

Comment:
This seems too large for a single sample? Same question about `hum_buf[64]` below.

Reply:
Changed to use sensor_q31_data struct instead of a buffer, take a look